packages/database-persistence-layer/README.md

```diff
@@ -0,0 +1,3 @@
+# Database persistence layer
```

Comment:
Thinking about the name here:

 - Should the name include "preferences" to clarify that it's about persisting preferences
 - Should the name include "api" instead of "database" (I think that's probably arguable)?

Reply:
> Should the name include "preferences" to clarify that it's about persisting preferences

I'd like to add data migration code to the package that's specific to preferences, so I think it makes sense to mention `preferences`.

A problem with `api` and `database` is that they're quite specific about the implementation. If we wanted to change it in future we'd probably need a new package.

We could go with something like `preferences-persistence-layer` and in the README state that this is the persistence layer for WordPress preferences. Maybe it exports `createUserMetaPersistenceLayer`, and if in the future we want to persist preferences in a different way, a new function can be added and the old one can be deprecated.

Reply:
the problem with `preferences-persistence-layer` is that it a local storage is also a persistence layer. I'm fine keeping "database" or "server" or something like that.

Reply:
> A problem with api and database is that those names are about the implementation. If we wanted to change it in future we'd probably need a new package.

It's hard to imagine that this implementation would change for WordPress in the future. Anyway, based on the keywords used in the code and your discussion I wanted to check what do you think about the following ideas:
- `@wordpress/user-preferences`
- `@wordpress/preferences-user-persistence`

Reply:
I quite like those two suggestions @gziolo. Do you have a preference @youknowriad?

Reply:
What about just `@wordpress/preferences-persistence` ?

Reply:
> You've chosen database which I think is a little misleading because it really uses the REST API. I doubt that will ever change.

I'm not so sure about that, a switch to GraphQL, websockets, or webRTC could all make that name irrelevant. It would be a very big change for WordPress, but then using a realtime communication protocol is very likely to become a reality for collaborative editing. While it's unlikely everything would go through such an API, I think it's enough of a reason not to consider 'REST API' in the naming.

It's this reason that most other packages are named for what they do rather than how they do it. There are exceptions, but most of those are very small utils. Because of that I would favour something like `preferences-persistence` or `preferences-user-persistence`.

I think I'll go with `preferences-persistence` as Riad suggested, just to keep it simple.